src/typecheck/mod.rs

```diff
@@ -2898,43 +2702,8 @@ pub fn constr_unify_rrows(
             UnifRecordRows::Concrete(RecordRowsF::Extend { tail, .. }) => {
                 constr_unify_rrows(constr, var_id, tail)
             }
-            UnifRecordRows::UnifVar(u) if *u != var_id => {
-                if let Some(u_constr) = constr.get_mut(u) {
-                    u_constr.extend(p_constr.into_iter());
-                } else {
-                    constr.insert(*u, p_constr);
-                }
-
-                Ok(())
-            }
-            _ => Ok(()),
-        }
-    } else {
-        Ok(())
-    }
-}
```

Comment:
Nothing to do with your PR, but this is a good example of when the git diff isn't very smart. I was trying to understand why you got rid of that part, but it's just an artifact of the next function being very similar and removed as well, makes it like your removed half this function and half the next one, instead of just the next one

Reply:
Yup. Very annoying. I actually got confused by exactly the same part of the diff at some point.